api/handler/copy.go

```diff
@@ -91,6 +91,17 @@ func (h *handler) CopyObjectHandler(w http.ResponseWriter, r *http.Request) {
 	}
 
 	if containsACL {
+		state, err := h.cachedACLGetter.GetState(r.Context(), dstBktInfo.CID)
+		if err != nil {
+			h.logAndSendError(w, "could not get bucket acl state", reqInfo, err)
+			return
+		}
+
+		if state == data.BucketACLBucketOwnerEnforced {
```

Comment:
Looks like you're not handling "PUT requests that specify bucket owner full control ACLs, such as the bucket-owner-full-control canned ACL" correctly.

Reply:
Updated